crates/stackable-operator/src/utils/kubelet.rs

```diff
@@ -0,0 +1,77 @@
+use http;
+use k8s_openapi::api::core::v1::Node;
+use kube::{
+    Api,
+    api::{ListParams, ResourceExt},
+    client::Client,
+};
+use serde::Deserialize;
+use snafu::{OptionExt, ResultExt, Snafu};
+
+use crate::commons::networking::DomainName;
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("failed to list nodes"))]
+    ListNodes { source: kube::Error },
+
+    #[snafu(display("failed to build \"/api/v1/nodes/{node_name}/proxy/configz\" request"))]
```

Comment:
Thanks for adding more details!
optional: Ideally we would not construct the URL twice (error message and code), but let the code construct the URL once and pass it to the error message if needed